cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/BungeeCloudNetHelper.java

```diff
@@ -44,8 +44,19 @@ public static int getLastOnlineCount() {
         return lastOnlineCount;
     }
 
-    public static boolean isOnAFallbackInstance(ProxiedPlayer proxiedPlayer) {
-        return proxiedPlayer.getServer() != null && isFallbackServer(proxiedPlayer.getServer().getInfo());
+    public static boolean isOnMatchingFallbackInstance(ProxiedPlayer player) {
+        ServerInfo currentServer = player.getServer() == null ? null : player.getServer().getInfo();
+
+        if (currentServer != null) {
+            return BridgeProxyHelper.filterPlayerFallbacks(
+                    player.getUniqueId(),
+                    currentServer.getName(),
+                    player::hasPermission
+            ).anyMatch(proxyFallback ->
+                    proxyFallback.getTask().equals(BridgeProxyHelper.getCachedServiceInfoSnapshot(currentServer.getName()).getServiceId().getTaskName()));
```

Comment:
Even if it shouldn't be the case, check if this ServiceInfo is null

Reply:
Maybe return an Optional ServiceInfoSnapshot